rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -367,11 +369,13 @@ Node::list_parameters(
     {
       result.names.push_back(kv.first);
       size_t last_separator = kv.first.find_last_of('.');
-      std::string prefix = kv.first.substr(0, last_separator);
-      if (std::find(result.prefixes.cbegin(), result.prefixes.cend(),
-        prefix) == result.prefixes.cend())
-      {
-        result.prefixes.push_back(prefix);
+      if (std::string::npos != last_separator) {
+        std::string prefix = kv.first.substr(0, last_separator);
+        if (std::find(result.prefixes.cbegin(), result.prefixes.cend(),
+          prefix) == result.prefixes.cend())
```

Comment:
I found reading this line pretty difficult due to the wrapping within the `find` call. Wrapping after the comparison operator `==` helped me to parse the line.